controllers/client/openstackclient_controller.go

```diff
@@ -307,17 +308,23 @@ func (r *OpenStackClientReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 
 	spec := openstackclient.ClientPodSpec(ctx, instance, helper, configVarsHash)
 
+	podSpecHash, err := util.ObjectHash(spec)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	podSpecAnnotation := labels.GetGroupLabel("openstackclient") + "/pod-spec-hash"
```

Comment:
Note: replicaset uses a label vs an annotation but a label is too short to store our object hash string. Replicaset uses a shorter hash type but then needs logic to handle hash collisions